source/common/upstream/health_discovery_service.h

```diff
@@ -21,18 +91,27 @@ namespace Upstream {
 // clang-format on
 
 /**
- * Struct definition for all load reporter stats. @see stats_macros.h
+ * Struct definition for all hds stats. @see stats_macros.h
  */
 struct HdsDelegateStats {
   ALL_HDS_STATS(GENERATE_COUNTER_STRUCT)
 };
 
+/**
+ * The HdsDelegate class is responsible for receiving requests from a management
+ * server with a set of hosts to healthcheck, healthchecking them, and reporting
+ * back the results.
+ */
 class HdsDelegate
```

Comment:
Can we add a TODO around adding `/config_dump` support for this? I think it would be extremely useful to be able to see what hosts we are health checking like we do for the other xDS endpoints. Also, might consider adding a TODO around whether we want to add health check clusters to the `/clusters` endpoint to get detailed stats about each HC host. Also I think would be extremely useful for debugging.